### PR TITLE
pml_ucx: fix tag/context_id layout and upper bounds.

### PR DESCRIPTION
**Pull request by yosefe**
picked from master #4141

### CODE REVIEW

**Comment by yosefe** [comment]
@jladd-mlnx @miked-mellanox

File: ompi/mca/pml/ucx/pml_ucx_request.h [hunk 1]
```diff
@@ -26,15 +26,15 @@ enum {
...
+ * 01234567 01234567 01234567 01234567 01234567 0123 4567 01234567 01234567
+ *                           |                      |
+ *      message tag (24)     |   source rank (20)   |     context id (20)
+ *                           |                      |
  */
 #define PML_UCX_TAG_BITS                       24
-#define PML_UCX_RANK_BITS                      24
-#define PML_UCX_CONTEXT_BITS                   16
-#define PML_UCX_ANY_SOURCE_MASK                0x800000000000fffful
+#define PML_UCX_RANK_BITS                      20
```
Comment (jladd-mlnx) [inline ompi/mca/pml/ucx/pml_ucx_request.h]: Only 20 bits for ranks? So UCX PML will only support ~1 million ranks?

Comment (yosefe) [reply ompi/mca/pml/ucx/pml_ucx_request.h]: yes

Comment (jladd-mlnx) [reply ompi/mca/pml/ucx/pml_ucx_request.h]: For exascale it's not enough. Can we extend it later?

**Pull request merged**